crates/subspace-service/src/dsn/import_blocks.rs

```diff
@@ -170,15 +178,34 @@ where
     Client: HeaderBackend<Block> + BlockBackend<Block> + Send + Sync + 'static,
     IQS: ImportQueueService<Block> + ?Sized,
 {
-    // TODO: Consider introducing and using global in-memory segment header cache (this comment is
-    //  in multiple files)
-    let segment_commitments = SegmentHeaderHandler::new(node.clone())
+    debug!("Waiting for connected peers...");
+    if tokio::time::timeout(WAIT_FOR_PEERS_TIMEOUT, node.wait_for_connected_peers())
```

Comment:
I would add `Option<Timeout>` to `node.wait_for_connected_peers())` instead. Because it's a useful feature.